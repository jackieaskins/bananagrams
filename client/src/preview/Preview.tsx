import { Board } from '../boards/types';
import { Hand } from '../hands/types';
import PreviewBoard from './PreviewBoard';
import PreviewHand from './PreviewHand';

type PreviewProps = {
  board: Board;
  hand: Hand;
};

const Preview = ({ board, hand }: PreviewProps): JSX.Element => (
  <div
    css={{
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      flexGrow: 1,
    }}
  >
    <PreviewBoard board={board} />
    <PreviewHand hand={hand} />
  </div>
);

export default Preview;
