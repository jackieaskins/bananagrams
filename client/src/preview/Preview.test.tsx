import { shallow } from 'enzyme';

import Preview from './Preview';

describe('<Preview />', () => {
  const renderComponent = (propOverrides = {}) =>
    shallow(<Preview board={{}} hand={[]} {...propOverrides} />);

  it('renders preview board and hand', () => {
    expect(renderComponent()).toMatchSnapshot();
  });
});
