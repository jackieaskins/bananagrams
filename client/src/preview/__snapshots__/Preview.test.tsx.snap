// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview /> renders preview board and hand 1`] = `
.emotion-0u2j8te-Preview {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="emotion-0"
>
  <PreviewBoard
    board={Object {}}
  />
  <PreviewHand
    hand={Array []}
  />
</div>
`;
